### PR TITLE
[196] Partial implementation of Expression while exporting as textual files

### PR DESCRIPTION
**Pull request by wbilem**
Bug:https://github.com/eclipse-syson/syson/issues/196

### CODE REVIEW

**Comment by AxelRICHARD** [comment]
Please check why the build fails and please indicate in your commit title that it is related to the export

**Review by adaussy — CHANGES_REQUESTED** [review]

File: backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java [hunk 1]
```diff
@@ -895,4 +913,102 @@ public void literalInfinity() {
...
+    /**
+     * Test model for AttributeUsage with BinaryOperatorExpression
+     *
+     * <pre>
+     * attribute attribute1 = bestFuelConsumption + idlingFuelConsumption * tpd_avg;
+     * </pre>
+     *
+     * @author wbilem
+     */
+    private static class AttributeUsageTestModel {
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java:926]: done

File: backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java [hunk 2]
```diff
@@ -895,4 +913,102 @@ public void literalInfinity() {
...
+        private final AttributeUsage idlingFuelConsumption;
+
+        private final AttributeUsage tpdAvg;
+
+        private final AttributeUsage mm;
+
+        AttributeUsageTestModel() {
+            this.attributeUsage = this.builder.createWithName(AttributeUsage.class, "attribute1");
+            this.bestFuelConsumption = this.builder.createWithName(AttributeUsage.class, "bestFuelConsumption");
+            this.idlingFuelConsumption = this.builder.createWithName(AttributeUsage.class, "idlingFuelConsumption");
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java:943]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 3]
```diff
@@ -379,14 +407,234 @@ private String mapToExpression(Element element, Feature feature) {
...
+    private void appendPrimaryExpression(Appender builder, Expression expression) {
+        if (expression instanceof FeatureChainExpression feature) {
+            LOGGER.warn("FeatureChainExpression are not handled yet");
+        } else {
+            this.appendNonFeatureChainExpression(builder, expression);
+        }
+    }
+
+    private void appendNonFeatureChainExpression(Appender builder, Expression expression) {
+        if (expression instanceof SelectExpression exp) {
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:499]: We leave it like that for now after discussion

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 4]
```diff
@@ -379,14 +407,234 @@ private String mapToExpression(Element element, Feature feature) {
...
+            LOGGER.warn("SelectExpression are not handled yet");
+        } else if (expression instanceof CollectExpression exp) {
+            LOGGER.warn("CollectExpression are not handled yet");
+        } else if (expression instanceof NullExpression exp) {
+            LOGGER.warn("NullExpression are not handled yet");
+        } else if (expression instanceof LiteralExpression exp) {
+            builder.appendSpaceIfNeeded().append(this.caseLiteralExpression(exp));
+        } else if (expression instanceof FeatureReferenceExpression exp) {
+            this.appendFeatureReferenceExpression(builder, exp);
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:508]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 5]
```diff
@@ -379,14 +407,234 @@ private String mapToExpression(Element element, Feature feature) {
...
+            } else {
+                builder.append("=");
+            }
+            this.appendOwnedExpression(builder, feature.getValue());
+        }
+
+    }
+
+    private void appendOwnedExpression(Appender builder, Expression expression) {
+        if (expression instanceof OperatorExpression op && op.getOperator() != null) {
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:435]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 6]
```diff
@@ -236,24 +250,38 @@ public String caseLiteralInteger(LiteralInteger literal) {
...
         this.appendChildrenContent(builder, usage, children);
 
         return builder.toString();
     }
 
+    @Override
+    public String caseFeatureReferenceExpression(FeatureReferenceExpression expression) {
+        Appender builder = this.newAppender();
+
+        expression.getOwnedRelationship().stream()
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:274]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 7]
```diff
@@ -139,7 +154,6 @@ public String caseNamespace(Namespace namespace) {
 
     @Override
     public String casePackage(Package pack) {
-
```
[the remark this answers is not in the document]
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:142]: done

File: backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java [hunk 8]
```diff
@@ -671,6 +673,20 @@ public void attributeUsage() {
         this.assertTextualFormEquals("attribute attribute1;", attributeUsage);
     }
 
+    @Test
+    public void attributeUsageWithBinaryOperatorExpression() {
+        AttributeUsage attributeUsage = this.builder.createWithName(AttributeUsage.class, ATTRIBUTE1);
```
Comment (adaussy) [inline backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java:678]: Why this part does not belong to the TestModel?

File: backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java [hunk 9]
```diff
@@ -671,6 +673,20 @@ public void attributeUsage() {
...
+        this.assertTextualFormEquals("attribute attribute1 = bestFuel + idlingFuel * fuel;", attributeUsage);
+    }
+
+    @Test
+    public void attributeUsageWithBracketOperatorExpression() {
+        AttributeUsage attributeUsage = this.builder.createWithName(AttributeUsage.class, ATTRIBUTE1);
+        new AttributeUsageWithBracketOperatorExpressionTestModel(attributeUsage);
+        this.assertTextualFormEquals("attribute attribute1 = 80 [millimetre];", attributeUsage);
+    }
+
```
Comment (adaussy) [inline backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java:689]: Why this part does not belong to the TestModel?

**Review by adaussy — CHANGES_REQUESTED** [review]

File: backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java (hunk 8 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/test/java/org/eclipse/syson/sysml/export/SysMLModelToTextSwitchTest.java]: done

**Comment by AxelRICHARD** [comment]
Please stop adding `[export]` or any other `[topic]`in commit/PR/ticket titles

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**